modules/GEPanelApp.py

```diff
@@ -0,0 +1,248 @@
+import re
+import csv
+import requests
+from ontoma import OnToma
+from settings import Config
+from collections import OrderedDict
+from common.HGNCParser import GeneParser
+from common.GCSUtils import GCSBucketManager
+
+import logging
+logger = logging.getLogger(__name__)
+
+__copyright__ = "Copyright 2014-2018, Open Targets"
+__credits__   = ["Gautier Koscielny", "ChuangKee Ong"]
+__license__   = "Apache 2.0"
+__version__   = "1.2.8"
+__maintainer__= "ChuangKee Ong"
+__email__     = ["data@opentargets.org"]
+__status__    = "Production"
+
+##TODO: 1) implement logging
+
+'''
+* Get 'Phenotypes' list for each 'HighEvidence' gene from each GE Panel.
+* For each phenotype 1st check that it has an OMIM id [6 digits].
+* If OMIM id found, try map to EFO using mapping file
+  OMIM_EFO_MAP => https://github.com/opentargets/OnToma/blob/master/ontoma/constants.py
+  ** If EFO mapping found
+    create a 'self.map_strings' with 'panel_name, gene_symbol, levelOfConfidence, phenotypes, efo_id'
+    i.e Neutropenia, severe congenital 3, autosomal recessive, 610738
+        OMIM  130650
+        #120330:Papillorenal syndrome
+  ** If NO EFO mapping found try OntoMA
+    i.e Verheij syndrome,  615583
+        => Ontoma: http://www.ebi.ac.uk/efo/EFO_0000699,
+* If NO OMIM id found, try OntoMa
+    i.e. Diffuse keratoderma with knuckle pads
+        => Ontoma: 'http://www.orpha.net/ORDO/Orphanet_2698'
+
+* The OntoMa mapping process
+  ** https://github.com/opentargets/OnToma/blob/master/ontoma/interface.py#L365
+  ** Searches for a matching EFO code for a given phenotype/disease string
+     operations roughly ordered from least expensive to most expensive
+     and also from most authorative to least authorative
+        1. EFO OBO lookup
+        2. Zooma mappings lookup
+        3. Zooma API high confidence lookup
+        4. OLS API EFO lookup - exact match
+        --- below this line we might not have a term in the platform ---
+        5. HP OBO lookup
+        6. OLS API HP lookup - exact match
+        7. OLS API EFO lookup - not exact
+        (8. ?Zooma medium)
+
+'''
+
+#class GEPanelApp(GCSBucketManager):
+class GEPanelApp():
+    #ge_object = GE()
+    #ge_object.execute_ge_request()
+    #ge_object.use_zooma()
+    #ge_object.process_panel_app_file()
+    #ge_object.write_evidence_strings(Config.GE_EVIDENCE_FILENAME)
+
+    def __init__(self):
+        super(GEPanelApp, self).__init__()
+        self.panel_app_id_map = OrderedDict()
+        self.panel_app_info = list()
+        self.map_strings = OrderedDict()
+
+    def process_ge(self):
+
+        self.ontoma = OnToma()
+        gene_parser = GeneParser()
+        gene_parser._get_hgnc_data_from_json()
+
+        self.omim_to_efo_map = self.ontoma._omim_to_efo
```

Comment:
ontoma already implements a `omim_lookup` https://github.com/opentargets/OnToma/blob/master/ontoma/interface.py#L259. shouldn't we use this instead of accessing the _omim_to_efo private object ?

using the function allows us to change the logic in the future if needed.